src/container/__tests__/RelayContainer_setVariables-test.js

```diff
@@ -292,6 +292,15 @@ describe('RelayContainer.setVariables', function() {
       expect(mockInstance.state.queryData.entity).toBe(updatedQueryData);
     });
 
+    it('re-renders when there are pendingVariables', () => {
+      const before = render.mock.calls.length;
+      expect(render.mock.calls.length).toBe(1);
+      mockInstance.setVariables({site: 'www'});
+      jest.runAllTimers();
+      environment.primeCache.mock.requests[0].block();
+      expect(render.mock.calls.length - before).toBe(1);
```

Comment:
@chollier so this test passes even w/o the changes to RelayContainer? if you log the arguments to `render`, what are they? try logging with e.g `console.log(render.mock.calls)` and see what `props.relay` is for each render.

you could also try calling `jest.runAllTimers()` *after* the `block()` on line 300 and see what that does (might cause another render)